crates/router/src/core/customers.rs

```diff
@@ -621,7 +621,7 @@ impl CustomerDeleteBridge for customers::GlobalId {
             // check this in review
             Ok(customer_payment_methods) => {
                 for pm in customer_payment_methods.into_iter() {
-                    if pm.payment_method == Some(enums::PaymentMethod::Card) {
+                    if pm.payment_method_type_v2 == Some(enums::PaymentMethod::Card) {
```

Comment:
This field is being accessed from the diesel model, which doesn't have methods to access these fields (only domain models do as of now). Let me know if I should add methods to the diesel models to access these fields as well.